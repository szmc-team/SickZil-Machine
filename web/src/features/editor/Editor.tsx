/** @jsx jsx */
import { jsx, css } from '@emotion/core'
import { useState } from 'react'
import { subject } from '../explorer/components/ImageListItem'
import { useFileEntryQuery } from '~/graphql'

const Editor: React.FC = () => {
  const [imgId, setImgId] = useState<string>('')

  const { data: fileEntryData } = useFileEntryQuery({
    variables: {
      id: imgId,
    },
  })
  const fileEntry = fileEntryData?.fileEntry

  subject.subscribe({
    next: (v: String) => {
      setImgId(v as string)
    },
  })

  return (
    <div css={styles.editor}>
      <label
        css={css`
          display: block;
          width: 100%;
          height: 100%;
        `}
      >
        <img
          css={css`
            width: 100%;
            height: 100%;
            object-fit: contain;
          `}
          src={fileEntry?.url}
        />
      </label>
    </div>
  )
}
const styles = {
  editor: css`
    width: 100%;
    background-color: black;
    overflow: hidden;
  `,
}

export default Editor
