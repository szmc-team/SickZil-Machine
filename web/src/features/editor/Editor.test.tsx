import React from 'react'
import { render, fireEvent } from '@testing-library/react'
import Editor from './Editor'

describe('<Editor />', () => {
  it('mathces snapshot', () => {
    const utils = render(<Editor />)
    expect(utils.container).toMatchSnapshot()
  })
  it('check canvas loading', () => {
    const utils = render(<Editor />)
    const input = utils.getByLabelText('editor_input')
    fireEvent.change(input, { target: { value: '' } })
    const image = utils.getByAltText('Image not found')
    expect(image).toBeInTheDocument
  })
})
