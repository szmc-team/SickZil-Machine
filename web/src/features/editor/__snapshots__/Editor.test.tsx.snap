// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Editor /> mathces snapshot 1`] = `
<div>
  <div
    class="css-1eoobr4-Editor"
  >
    <label
      class="css-dmhmyr-Editor"
    >
      <input
        accept="image/*"
        aria-label="editor_input"
        class="css-mmdtz1-Editor"
        type="file"
      />
      <div
        class="css-15ujusl-Editor"
      >
        <img
          alt="Image not found"
          class="css-kngyu6-Editor"
          src=""
        />
      </div>
    </label>
  </div>
</div>
`;
